### DISCUSSION TITLE
prettier-plugin not working with sort imports plugin

### DISCUSSION THREAD

**Discussion by Omar-Elmassry** [tailwindlabs/tailwindcss#13001] [Help]
Hi 
I'm using the prettier plugin in one of my projects, it was working great until i decided to use [import sorting plugin](https://github.com/IanVS/prettier-plugin-sort-imports) then it will not work at all.
i tried commenting the sort imports plugin in my `prettier.config,js` and it worked again.

- "prettier": "^3.2.5"
- "prettier-plugin-tailwindcss": "^0.5.11",

here is my prettier.config,js
```
export default {
  endOfLine: "lf",
  semi: true,
  singleQuote: false,
  tabWidth: 2,
  useTabs: false,
  trailingComma: "es5",

  importOrder: [
    "^(react/(.*)$)|^(react$)",
    "^(react-dom/(.*)$)|^(react-dom$)",
    "^(react-router-dom/(.*)$)|^(react-router-dom$)",
    "<THIRD_PARTY_MODULES>",
    "",
    "<TYPES>",
    "<TYPES>^[.]",
    "^@types/(.*)$",
    "",
    "^@lib/(.*)$",
    "^@hooks/(.*)$",
    "^@store/(.*)$",
    "^@auth/(.*)$",
    "",
    "^[@/]",
    "^@assets/(.*)$",
    "^@pages/(.*)$",
    "^@components/ui/(.*)$",
    "^@components/(.*)$",
    "",
    "^[.]",
  ],
  importOrderParserPlugins: ["typescript", "jsx", "decorators-legacy"],

  plugins: [
    "prettier-plugin-tailwindcss",
    "@ianvs/prettier-plugin-sort-imports",
  ],
  tailwindFunctions: ["cva", "clsx", "cn"],
};
```

any help will be appreciated and thanks alot for tremendous efforts.

**Comment by shabazs453** [comment]
Hi there,

experiencing the same issue. The one which is declared later works and other not.
This is my file of prettier configuration

{
  "endOfLine": "lf",
  "singleQuote": false,
  "trailingComma": "es5",
  "importOrder": [
    "^(react/(.*)$)|^(react$)",
    "^(next/(.*)$)|^(next$)",
    "<THIRD_PARTY_MODULES>",
    "",
    "^types$",
    "^@/types/(.*)$",
    "^@/env(.*)$",
    "^@/config/(.*)$",
    "^@/lib/(.*)$",
    "^@hooks/(.*)$",
    "^@/components/ui/(.*)$",
    "^@/components/(.*)$",
    "^@/styles/(.*)$",
    "^@/app/(.*)$",
    "",
    "^[./]",
    "^(?!.*[.]css$)[./].*$",
    ".css$"
  ],
  "importOrderSeparation": true,
  "importOrderSortSpecifiers": true,
  "importOrderMergeDuplicateImports": true,
  "tailwindFunctions": ["clsx", "cn", "cva"],
  "plugins": [
    "prettier-plugin-tailwindcss",
    "@ianvs/prettier-plugin-sort-imports"
  ]
}


How is it possible that no one in years have single solution for the issue on the web, nor stack overflow or youtube?
Although the official documentation suggests to declare prettier-plugin-tailwindcss at the last of plugins list as per **[](https://github.com/tailwindlabs/prettier-plugin-tailwindcss)**

But it is not working for me either
